homeassistant/components/zha/core/registries.py

```diff
@@ -405,10 +404,12 @@ def decorator(zha_entity: CALLABLE_T) -> CALLABLE_T:
 
         return decorator
 
-    def group_match(self, component: str) -> Callable[[CALLABLE_T], CALLABLE_T]:
+    def group_match(
+        self, component: str
+    ) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
         """Decorate a group match rule."""
 
-        def decorator(zha_ent: CALLABLE_T) -> CALLABLE_T:
+        def decorator(zha_ent: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
This one is slightly different as it's for groups. Instead of `_ZhaEntityT`, I used
```py
_ZhaGroupEntityT = TypeVar("_ZhaGroupEntityT", bound=type["ZhaGroupEntity"])
```